### ISSUE TITLE
Adding custom loss function

### ISSUE THREAD

**Issue by subho22** [jeroenvdhoven/mlpype#83]
How can we add a custom loss function for training a model using this package ?

**Comment by jeroenvdhoven** [comment]
Hey, thanks for sharing an interest in this package :)

Adding a loss function depends a bit on your use. I assume you're looking to use some loss function for training a neural network. For Keras models, you can use a regular Keras-compatible loss function for the `loss` argument in a [KerasPypeModel](https://github.com/jeroenvdhoven/mlpype/blob/master/mlpype-tensorflow/mlpype/tensorflow/model/keras_pype_model.py). An example of how to use a Keras model is given [here](https://github.com/jeroenvdhoven/mlpype/blob/master/examples/tensorflow/keras_model_example.py).

If you're mainly interested in just evaluating some extra metrics afterwards, you can add custom functions to the Evaluator needed when you create an Experiment. See for instance [the sklearn example](https://github.com/jeroenvdhoven/mlpype/blob/master/examples/sklearn/sklearn_example.py) for how the `accuracy_score` from sklearn is included. The code for the Evaluator can be found [here](https://github.com/jeroenvdhoven/mlpype/blob/master/mlpype-base/mlpype/base/evaluate/evaluator.py)

If you have any more questions, I'll be more than willing to help out :)

**Comment by subho22** [comment]
Hey for using xgbregressor using mlpype if I want to add or pass a custom loss function which has been created by my own. What changes should be done in the code ?

**Comment by jeroenvdhoven** [comment]
Hey, those are based on the base [Sklearn Model implementation](https://github.com/jeroenvdhoven/mlpype/blob/master/mlpype-sklearn/mlpype/sklearn/model/sklearn_model.py). All kwargs are passed as you'd expect to the XGBregressor, so going by [their documentation](https://xgboost.readthedocs.io/en/stable/python/python_api.html#module-xgboost.sklearn), you should be able to do:

```python
from mlpype.xgboost.model import XGBRegressorModel

XGBRegressorModel(
    inputs=[...],  # Whatever input names you need
    outputs=[...],  # Whatever output names you need
    # extra arguments can be passed through the kwargs
    n_estimators=20,
    objective=<your custom loss function here>,
    max_depth=3,
)
```

Thanks for letting me know, I noticed with some recent changes the docs for these models aren't in a good state. Will work to fix that.

**Comment by subho22** [comment]
Also the custom objective function should take arguments in a certain format like predt: np.ndarray, dtrain: xgb.DMatrix.
Will this work in here as well ?

**Comment by jeroenvdhoven** [comment]
All `kwargs` are passed directly to the XGBRegressor class from xgboost itself, mlpype doesn't do anything with them. MLpype makes sure the data is presented to the `fit` method of `XGBRegressor` such that it can do its job of training the model. So indeed, that should work. If you're running into any issues with the custom objective, could you share some code so I can take a look and debug it?
